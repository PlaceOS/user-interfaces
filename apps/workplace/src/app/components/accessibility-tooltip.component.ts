import { Component, Inject } from '@angular/core';
import { ApplicationLink, SettingsService } from '@placeos/common';
import { CustomTooltipData } from '@placeos/components';

@Component({
    selector: 'accessibility-tooltip',
    template: `
        <div
            class="flex flex-col w-[18.5rem] rounded bg-white shadow relative -top-12 -right-1 overflow-hidden dark:bg-neutral-700 dark:text-white"
        >
            <div class="flex items-center space-x-2 p-2" (click)="close()">
                <app-icon class="text-2xl">arrow_back</app-icon>
                <div class="" i18n>Display & Accessibility</div>
            </div>
            <div
                action
                class="flex items-center space-x-2 w-full text-left h-auto px-4"
            >
                <div
                    class="flex items-center justify-center rounded-full w-8 h-8 bg-gray-200 dark:bg-neutral-800 my-2"
                >
                    <app-icon class="text-xl">mode_night</app-icon>
                </div>
                <div class="flex-1 text-sm" i18n>Dark Mode</div>
                <mat-slide-toggle
                    [ngModel]="dark_mode"
                    (ngModelChange)="applySetting('dark_mode', $event)"
                ></mat-slide-toggle>
            </div>
            <div
                action
                class="flex items-center space-x-2 w-full text-left h-auto px-4"
            >
                <div
                    class="flex items-center justify-center rounded-full w-8 h-8 bg-gray-200 dark:bg-neutral-800 my-2"
                >
                    <app-icon class="text-xl">playlist_add</app-icon>
                </div>
                <div class="flex-1 text-sm" i18n>Large Accessibility Sizes</div>
                <mat-slide-toggle
                    [ngModel]="accessible"
                    (ngModelChange)="applySetting('accessible', $event)"
                ></mat-slide-toggle>
            </div>
            <ng-container *ngIf="accessible">
                <div class="px-8 py-4 bg-gray-200 dark:bg-neutral-600 text-center" i18n>
                    Adjust your preferred reading size below.
                </div>
                <div class="flex items-center px-4 space-x-4">
                    <span class="text-sm">A</span>
                    <mat-slider
                        class="flex-1 w-1/2"
                        [min]="10"
                        [max]="22"
                        [ngModel]="font_size"
                        (ngModelChange)="applySetting('font_size', $event)"
                        [step]="2"
                    ></mat-slider>
                    <span class="text-2xl">A</span>
                </div>
            </ng-container>
        </div>
    `,
    styles: [
        `
            :host > div > [action] {
                border-top: 1px solid #ccc;
            }
        `,
    ],
})
export class AccessibilityTooltipComponent {

    public get dark_mode() {
        return !!this._settings.get('dark_mode');
    }

    public get accessible() {
        return !!this._settings.get('accessible');
    }

    public get font_size() {
        return this._settings.get('font_size') || 16;
    }

    public readonly applySetting = (n, v) =>
        this._settings.saveUserSetting(n, v);

    public readonly close = () => this._data?.close();

    constructor(
        @Inject(CustomTooltipData) private _data: any,
        private _settings: SettingsService
    ) {}
}
