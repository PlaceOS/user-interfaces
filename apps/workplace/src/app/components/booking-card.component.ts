import { Component, Input } from '@angular/core';
import { Booking } from '@placeos/bookings';
import { OrganisationService } from '@placeos/organisation';
import { addMinutes, format, formatDuration, isSameDay } from 'date-fns';

@Component({
    selector: 'booking-card',
    template: `
        <h4 class="mb-2 flex items-center" *ngIf="booking">
            <span *ngIf="show_day">{{ day }}, </span>
            {{ booking?.date | date: 'h:mm a' }}
            <span class="text-xs px-2">({{ booking?.date | date:'z' }})</span>
        </h4>
        <a
            matRippleColor
            class="w-full"
            [routerLink]="['/schedule', 'view', booking?.id, 'booking']"
            *ngIf="booking"
        >
            <div class="w-full bg-white rounded shadow py-2 relative">
                <h4 class="px-2">{{ booking?.title }}</h4>
                <div class="flex m-2">
                    <div
                        class="flex items-center bg-opacity-30 rounded-2xl p-1 text-sm space-x-2 pr-2 font-medium"
                        [class.bg-success]="booking?.status === 'approved'"
                        [class.bg-pending]="booking?.status === 'tentative'"
                        [class.bg-error]="booking?.status === 'declined'"
                        [class.bg-gray-200]="!booking?.status"
                    >
                        <div
                            class="rounded-full h-5 w-5 flex items-center justify-center text-white"
                            [class.bg-success]="booking?.status === 'approved'"
                            [class.bg-pending]="booking?.status === 'tentative'"
                            [class.bg-error]="booking?.status === 'declined'"
                            [class.bg-gray-200]="!booking?.status"
                        >
                            <app-icon>
                                {{
                                    booking?.status === 'approved'
                                        ? 'done'
                                        : booking?.status === 'tentative'
                                        ? 'warning'
                                        : 'close'
                                }}
                            </app-icon>
                        </div>
                        <div class="pr-1">{{ period }}</div>
                    </div>
                </div>
                <div
                    class="flex flex-wrap flex-col sm:flex-row sm:divide-x divide-gray-200 py-2 space-y-2 sm:space-y-0"
                >
                    <div class="flex items-center px-2">
                        <app-icon *ngIf="type !== 'desk'; else desk_icon">{{ type }}</app-icon>
                        <div class="mx-2 truncate">
                            {{ booking?.description || booking?.asset_id }}
                        </div>
                    </div>
                    <div class="flex items-center px-2" *ngIf="location">
                        <app-icon>place</app-icon>
                        <div class="mx-2 truncate">{{ location }}</div>
                    </div>
                </div>
                <app-icon
                    class="absolute top-1/2 right-1 text-4xl -translate-y-1/2"
                >
                    chevron_right
                </app-icon>
            </div>
        </a>
        <ng-template #desk_icon>
            <img src="assets/icons/desk-outline.svg" class="w-5 h-5" />
        </ng-template>
    `,
    styles: [
        `
            :host {
                display: block;
                width: 100%;
            }
        `,
    ],
})
export class BookingCardComponent {
    @Input() public booking: Booking;
    @Input() public show_day: boolean = false;

    public get type() {
        if (this.booking?.type === 'desk') return 'desk';
        if (this.booking?.type === 'parking') return 'drive_eta';
        return 'book';
    }

    public get day() {
        const date = this.booking?.date || Date.now();
        const is_today = isSameDay(Date.now(), date);
        return `${is_today ? 'Today' : format(date, 'EEEE')}`;
    }

    public get location() {
        const level = this._org.levelWithID(this.booking?.zones || []);
        return `${level?.display_name || level?.name || ''}`;
    }

    public get period() {
        const start = this.booking?.date || Date.now();
        const duration = this.booking?.duration || 60;
        const end = addMinutes(start, duration);
        const dur = formatDuration({
            hours: Math.floor(duration / 60),
            minutes: duration % 60,
        })
            .replace(' hour', 'hr')
            .replace(' minute', 'min');
        return `${format(start, 'h:mm a')} - ${format(end, 'h:mm a')} (${dur})`;
    }

    constructor(private _org: OrganisationService) {}
}
