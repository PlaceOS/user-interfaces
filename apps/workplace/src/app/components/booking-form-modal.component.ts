import { Component, Input, OnInit } from '@angular/core';
import { FormGroup } from '@angular/forms';

import { BaseClass } from '@placeos/common';

@Component({
    selector: 'a-booking-form-modal',
    template: `
        <div class="booking-form" [formGroup]="form">
            <div *ngIf="first_form">
                <div>
                    <label>
                        <strong>Meeting Subject:<span>*</span></strong>
                    </label>
                    <mat-form-field appearance="outline">
                        <input
                            matInput
                            name="title"
                            placeholder="Enter here"
                            formControlName="title"
                        />
                    </mat-form-field>
                </div>
                <div *ngIf="form.controls.organiser">
                    <label for="organiser">
                        <strong>Host:<span>*</span></strong>
                    </label>
                    <a-user-search-field
                        name="organiser"
                        formControlName="organiser"
                    ></a-user-search-field>
                </div>
                <div>
                    <label for="start-time">
                        <strong>Time:<span>*</span></strong>
                    </label>
                    <a-time-field
                        name="start-time"
                        formControlName="date"
                    ></a-time-field>
                </div>
                <div style="margin-top: 1.5em;">
                    <label for="duration">
                        <strong>Duration:<span>*</span></strong>
                    </label>
                    <a-duration-field
                        name="duration"
                        formControlName="duration"
                    ></a-duration-field>
                </div>
            </div>
            <div [hidden]="first_form">
                <input
                    *ngIf="form.controls.notes"
                    class="notes"
                    matInput
                    name="notes"
                    placeholder="Add notes here..."
                    formControlName="notes"
                />
                <div *ngIf="form.controls.head_count">
                    <label for="organiser">
                        <strong>Estimated Headcount:<span>*</span></strong>
                    </label>
                    <mat-form-field appearance="outline">
                        <input
                            matInput
                            name="headcount"
                            placeholder="Expected attendees here.."
                            formControlName="head_count"
                        />
                    </mat-form-field>
                </div>
            </div>
        </div>
    `,
    styles: [
        `
            .field {
                display: flex;
                flex-wrap: wrap;
            }

            .notes {
                padding-bottom: 150px;
                width: 100%;
                background-color: lightgrey;
                border: solid 1px rgba(black, 0.5);
                margin-bottom: 50px;
            }

            mat-form-field {
                width: 100%;
            }
        `,
    ],
})
export class BookingFormModalComponent extends BaseClass {
    @Input() public first_form: boolean;
    @Input() public form: FormGroup;
}
