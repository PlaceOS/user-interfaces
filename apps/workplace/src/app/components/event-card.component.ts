import { Component, Input } from '@angular/core';
import { CalendarEvent } from '@placeos/events';
import { addMinutes, format, formatDuration, isSameDay } from 'date-fns';

@Component({
    selector: 'event-card',
    template: `
        <h4 class="mb-2 flex items-center" *ngIf="event" date>
            <span *ngIf="show_day" day>{{ day }}, </span>
            {{ event?.date | date: 'h:mm a' }}
            <span class="text-xs px-2">({{ event?.date | date: 'z' }})</span>
        </h4>
        <a
            matRippleColor
            details
            class="w-full"
            [routerLink]="['/schedule', 'view', event?.id, 'event']"
            *ngIf="event"
        >
            <div class="w-full bg-white rounded shadow py-2 relative">
                <h4 class="px-2">{{ event?.title }}</h4>
                <div class="flex m-2">
                    <div
                        class="flex items-center bg-opacity-30 rounded-2xl p-1 text-sm space-x-2 pr-2 font-medium"
                        [class.bg-green-600]="event?.status === 'approved'"
                        [class.bg-yellow-500]="event?.status === 'tentative'"
                        [class.bg-red-600]="event?.status === 'declined'"
                        [class.bg-gray-200]="!event?.status"
                    >
                        <div
                            class="rounded-full h-5 w-5 flex items-center justify-center text-white"
                            [class.bg-success]="event?.status === 'approved'"
                            [class.bg-pending]="event?.status === 'tentative'"
                            [class.bg-error]="event?.status === 'declined'"
                            [class.bg-gray-200]="!event?.status"
                        >
                            <app-icon>
                                {{
                                    event?.status === 'approved'
                                        ? 'done'
                                        : event?.status === 'tentative'
                                        ? 'warning'
                                        : 'close'
                                }}
                            </app-icon>
                        </div>
                        <div class="pr-1">{{ period }}</div>
                    </div>
                </div>
                <div
                    class="flex flex-wrap flex-col sm:flex-row sm:divide-x divide-gray-200 py-2 space-y-2 sm:space-y-0"
                >
                    <div class="flex items-center px-2">
                        <app-icon>meeting_room</app-icon>
                        <div class="mx-2 truncate">
                            {{
                                event?.space?.level?.display_name ||
                                    event?.space?.level?.name
                            }},
                            {{
                                event?.space?.display_name || event?.space?.name
                            }}
                        </div>
                    </div>
                    <div class="flex items-center px-2" *ngIf="location">
                        <app-icon>place</app-icon>
                        <div class="mx-2 truncate">{{ location }}</div>
                    </div>
                    <div class="flex items-center px-2">
                        <app-icon>person_outline</app-icon>
                        <div class="mx-2">
                            {{
                                event?.organiser?.name ||
                                    event?.organiser?.email
                            }}
                        </div>
                    </div>
                    <div
                        class="flex items-center px-2"
                        *ngIf="event?.ext('catering')?.length"
                    >
                        <app-icon>restaurant</app-icon>
                        <div class="mx-2">Catered</div>
                    </div>
                    <div class="flex items-center px-2">
                        <app-icon>people</app-icon>
                        <div class="mx-2">
                            {{ event?.attendees?.length }}
                            {{
                                event?.attendees?.length === 1
                                    ? 'Person'
                                    : 'People'
                            }}
                        </div>
                    </div>
                </div>
                <app-icon
                    class="absolute top-1/2 right-1 text-4xl -translate-y-1/2"
                >
                    chevron_right
                </app-icon>
                <div
                    class="absolute bottom-2 right-2 sm:bottom-auto sm:top-2 text-sm sm:text-base flex items-center pr-4"
                    *ngIf="event?.attendees?.length"
                >
                    <div
                        class="h-10 w-6"
                        *ngFor="
                            let user of event?.attendees
                                | slice
                                    : 0
                                    : (event?.attendees?.length === 6 ? 6 : 5)
                        "
                    >
                        <a-user-avatar [user]="user"></a-user-avatar>
                    </div>
                    <div class="h-10 w-6" *ngIf="event?.attendees?.length > 6">
                        <div
                            class="bg-secondary rounded-full h-10 w-10 border-2 border-white flex items-center justify-center text-white"
                        >
                            +{{ event?.attendees?.length - 5 }}
                        </div>
                    </div>
                </div>
            </div>
        </a>
    `,
    styles: [
        `
            :host {
                display: block;
                width: 100%;
            }
        `,
    ],
})
export class EventCardComponent {
    @Input() public event: CalendarEvent;
    @Input() public show_day: boolean = false;

    public get day() {
        const date = this.event?.date || Date.now();
        const is_today = isSameDay(Date.now(), date);
        return `${is_today ? 'Today' : format(date, 'EEEE')}`;
    }

    public get location() {
        return '';
    }

    public get period() {
        const start = this.event?.date || Date.now();
        const duration = this.event?.duration || 60;
        const end = addMinutes(start, duration);
        const dur = formatDuration({
            hours: Math.floor(duration / 60),
            minutes: duration % 60,
        })
            .replace(' hour', 'hr')
            .replace(' minute', 'min');
        return `${format(start, 'h:mm a')} - ${format(end, 'h:mm a')} (${dur})`;
    }
}
