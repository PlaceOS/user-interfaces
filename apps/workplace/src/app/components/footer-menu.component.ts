import { Component } from '@angular/core';
import { ApplicationLink, SettingsService } from '@placeos/common';

@Component({
    selector: 'footer-menu',
    template: `
        <div
            class="fixed z-20 inset-0 bottom-16 text-white"
            *ngIf="show_book_items"
            [attr.dark]="dark_mode"
            (click)="show_book_items = false"
        >
            <div class="absolute inset-0 bg-black opacity-50"></div>
            <div
                class="absolute inset-x-0 bottom-0 flex flex-wrap items-end justify-center p-4"
            >
                <a
                    name="footer-nav-meeting"
                    matRipple
                    [routerLink]="
                        new_features
                            ? ['/book', 'meeting']
                            : ['/book', 'spaces']
                    "
                    *ngIf="features.includes('spaces')"
                    routerLinkActive="active"
                    class="flex flex-col items-center justify-center space-y-2 w-1/3 m-2"
                >
                    <div
                        class="bg-base-100 text-base-content rounded-full h-12 w-12 text-2xl flex items-center justify-center"
                    >
                        <app-icon filled>meeting_room</app-icon>
                        <app-icon
                            outline
                            class="text-neutral"
                            className="material-icons-outlined"
                            >meeting_room</app-icon
                        >
                    </div>
                    <div class="text-xs" i18n>Book Meeting</div>
                </a>
                <a
                    matRipple
                    name="footer-nav-desks"
                    [routerLink]="
                        new_features
                            ? ['/book', 'new-desks']
                            : ['/book', 'desks']
                    "
                    routerLinkActive="active"
                    *ngIf="features.includes('desks')"
                    class="flex flex-col items-center justify-center space-y-2 w-1/3 m-2"
                >
                    <div
                        class="bg-base-100 text-base-content rounded-full h-12 w-12 text-2xl flex items-center justify-center"
                    >
                        <app-icon filled>desk</app-icon>
                        <app-icon
                            outline
                            class="text-neutral"
                            className="material-icons-outlined"
                            >desk</app-icon
                        >
                    </div>
                    <div class="text-xs" i18n>Book Desk</div>
                </a>
                <a
                    matRipple
                    name="footer-nav-parking"
                    [routerLink]="['/book', 'new-parking']"
                    routerLinkActive="active"
                    *ngIf="features.includes('parking')"
                    class="flex flex-col items-center justify-center space-y-2 text-base w-1/3 m-2"
                >
                    <div
                        class="bg-base-100 text-base-content rounded-full h-12 w-12 text-2xl flex items-center justify-center"
                    >
                        <app-icon filled>directions_car</app-icon>
                        <app-icon
                            outline
                            class="text-neutral"
                            className="material-icons-outlined"
                            >directions_car</app-icon
                        >
                    </div>
                    <div class="text-xs" i18n>Book Car Space</div>
                </a>
                <a
                    matRipple
                    name="footer-nav-visitor-invite"
                    *ngIf="features.includes('visitor-invite')"
                    [routerLink]="['/book', 'visitor']"
                    routerLinkActive="active"
                    class="flex flex-col items-center justify-center space-y-2 w-1/3 m-2"
                >
                    <div
                        class="bg-base-100 text-base-content rounded-full h-12 w-12 text-2xl flex items-center justify-center"
                    >
                        <app-icon filled>person_add</app-icon>
                        <app-icon
                            outline
                            class="text-neutral"
                            className="material-icons-outlined"
                            >person_add</app-icon
                        >
                    </div>
                    <div class="text-xs" i18n>Invite Visitor</div>
                </a>
                <a
                    matRipple
                    name="footer-nav-my-day"
                    [routerLink]="
                        new_features ? ['/your-bookings'] : ['/schedule']
                    "
                    *ngIf="features.includes('schedule')"
                    routerLinkActive="active"
                    class="flex flex-col items-center justify-center space-y-2 text-base w-1/3  m-2"
                >
                    <div
                        class="bg-base-100 text-base-content rounded-full h-12 w-12 text-2xl flex items-center justify-center"
                    >
                        <app-icon filled>today</app-icon>
                        <app-icon
                            outline
                            class="text-neutral"
                            className="material-icons-outlined"
                            >today</app-icon
                        >
                    </div>
                    <div class="text-xs" i18n>Your Bookings</div>
                </a>
            </div>
        </div>
        <div
            class="flex items-center justify-center bg-base-100 border-t border-base-200 shadow relative h-16 w-full sm:hidden z-40"
            *ngIf="features.length > 3; else simple_state"
            [attr.dark]="dark_mode"
        >
            <a
                matRipple
                class="flex flex-col items-center justify-center relative flex-1"
                [routerLink]="[default_page]"
                name="footer-nav-home"
                routerLinkActive="text-secondary active"
            >
                <app-icon filled class="text-2xl">home</app-icon>
                <app-icon
                    outline
                    className="material-icons-outlined"
                    class="text-2xl !m-0"
                    >home</app-icon
                >
                <span class="text-sm" i18n>Home</span>
            </a>
            <button
                matRipple
                class="flex items-center justify-center w-12 h-12 mb-4 rounded-full z-10"
                (click)="show_book_items = !show_book_items"
                [class.bg-secondary]="show_book_items"
                [class.text-white]="show_book_items"
                [class.bg-base-200]="!show_book_items"
            >
                <app-icon class="text-2xl">{{
                    show_book_items ? 'close' : 'add'
                }}</app-icon>
            </button>
            <a
                matRipple
                class="flex flex-col items-center justify-center relative flex-1"
                [routerLink]="['/explore']"
                routerLinkActive="text-secondary active"
            >
                <app-icon filled class="text-2xl">place</app-icon>
                <app-icon
                    outline
                    className="material-icons-outlined"
                    class="text-2xl !m-0"
                    >place</app-icon
                >
                <span class="text-sm" i18n>Spaces</span>
            </a>
            <div
                class="overflow-hidden absolute top-0 left-1/2 -translate-x-1/2 -translate-y-full w-24 h-2"
            >
                <div
                    class="w-16 h-16 rounded-full bg-base-100 shadow mx-auto border-t border-base-200"
                ></div>
            </div>
        </div>
        <ng-template #simple_state>
            <div
                class="flex items-center justify-center bg-base-100 border-t border-base-200 shadow relative h-16 w-full sm:hidden z-40"
            >
                <top-menu></top-menu>
            </div>
        </ng-template>
    `,
    styles: [
        `
            a.active app-icon {
                color: var(--s) !important;
            }

            a.active {
                font-weight: 500;
            }

            a:not(.active) [filled],
            a.active [outline] {
                display: none;
            }
        `,
    ],
})
export class FooterMenuComponent {
    public show_book_items = false;

    public get default_page(): string {
        return this._settings.get('app.default_route') || '/dashboard';
    }

    public get features(): string[] {
        return this._settings.get('app.features') || [];
    }

    public get new_features(): boolean {
        return !!this._settings.get('app.new_features');
    }

    public get dark_mode(): boolean {
        return !!this._settings.get('dark_mode');
    }

    constructor(private _settings: SettingsService) {}
}
