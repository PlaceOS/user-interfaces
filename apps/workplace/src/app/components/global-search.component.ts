import { Component } from '@angular/core';

import { ExploreSearchService } from '@placeos/explore';

@Component({
    selector: 'global-search',
    template: `
        <div
            container
            class="flex items-center rounded focus-within:shadow focus:shadow w-full px-2 relative"
        >
            <app-icon class="text-2xl">search</app-icon>
            <input
                #input
                placeholder="Search for people or spaces..."
                class="flex-1 w-1/2 py-2 outline-none"
                [(ngModel)]="filter_str"
                (ngModelChange)="setFilter($event)"
                [matAutocomplete]="auto"
                [matAutocompleteConnectedTo]="origin"
            />
            <mat-spinner *ngIf="loading | async" [diameter]="32"></mat-spinner>
            <div
                class="absolute bottom-0 left-2 right-2 min-w-[20rem]"
                matAutocompleteOrigin
                #origin="matAutocompleteOrigin"
            ></div>
        </div>
        <mat-autocomplete #auto="matAutocomplete">
            <ng-container *ngIf="(loading | async) !== true && filter_str">
                <mat-option
                    *ngIf="!(results | async)?.length"
                    class="pointer-events-none"
                >
                    No matches found
                </mat-option>
                <mat-option
                    *ngFor="let option of results | async | slice: 0:5"
                    [value]="option"
                >
                    <a
                        [routerLink]="['/explore']"
                        [queryParams]="
                            option.type === 'space'
                                ? { space: option.id }
                                : { email: option.id }
                        "
                        class="w-full h-full flex items-center leading-tight"
                    >
                        <div class="flex-1 overflow-hidden">
                            <div class="truncate w-full">
                                {{ option.name }}
                            </div>
                            <div class="text-xs">
                                {{ option.description }}
                            </div>
                        </div>
                        <div
                            class="text-xs font-bold p-2 capitalize text-white bg-gray-500 rounded"
                        >
                            {{ option.type }}
                        </div>
                    </a>
                </mat-option>
            </ng-container>
        </mat-autocomplete>
    `,
    styles: [
        `
            [container] {
                transition: color 200ms, background-color 200ms;
                background-color: #fff4;
                color: #fff8;
            }

            [container]:focus,
            [container]:focus-within {
                background-color: #fff;
                color: #000;
                outline: var(--primary);
            }

            input::placeholder {
                color: #fff8;
            }

            input:focus ::placeholder {
                color: #0008;
            }
        `,
    ],
})
export class GlobalSearchComponent {
    public readonly results = this._service.search_results;
    public readonly loading = this._service.loading;

    public filter_str = '';

    public readonly setFilter = (s) =>
        s instanceof Object
            ? (this.filter_str = '')
            : this._service.setFilter(s);

    constructor(private _service: ExploreSearchService) {}
}
