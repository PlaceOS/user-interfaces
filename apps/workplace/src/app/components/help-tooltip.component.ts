import { Component, Inject } from '@angular/core';
import { ApplicationLink, SettingsService } from '@placeos/common';
import { CustomTooltipData } from '@placeos/components';

@Component({
    selector: 'help-tooltip',
    template: `
        <div
            class="flex flex-col w-[18.5rem] rounded bg-white shadow relative -top-12 -right-1 dark:bg-neutral-700 dark:text-white divide-y divide-gray-300 dark:divide-neutral-500"
            (click)="close()"
        >
            <div
                class="flex items-center space-x-2 p-2 border-b dark:border-neutral-500"
            >
                <app-icon class="text-2xl">arrow_back</app-icon>
                <div class="" i18n>Help & Support</div>
            </div>
            <a
                btn
                matRipple
                *ngFor="let tile of tiles"
                class="clear w-full text-left h-auto"
                [href]="tile.link"
            >
                <div
                    class="flex items-center space-x-2 dark:text-white py-2 w-full"
                >
                    <div
                        class="flex items-center justify-center rounded-full w-8 h-8 bg-gray-200 dark:bg-neutral-800"
                    >
                        <app-icon class="text-xl" [icon]="tile.icon"></app-icon>
                    </div>
                    <div class="flex-1">{{ tile.name }}</div>
                </div>
            </a>
        </div>
    `,
    styles: [
        `
            a:not(:last-child) {
                border-bottom-width: 1px !important;
                border-bottom-style: solid !important;
                border-radius: 0;
            }
        `,
    ],
})
export class HelpTooltipComponent {
    /** Tiles to display on the help page */
    public get tiles(): ApplicationLink[] {
        return this._settings.get('app.help.tiles') || [];
    }

    public readonly close = () => this._data?.close();

    constructor(
        @Inject(CustomTooltipData) private _data: any,
        private _settings: SettingsService
    ) {}
}
