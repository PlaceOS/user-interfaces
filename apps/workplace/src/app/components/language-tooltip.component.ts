import { Component, Inject } from '@angular/core';
import { SettingsService } from '@placeos/common';
import { CustomTooltipData } from '@placeos/components';
import { OrganisationService } from '@placeos/organisation';

@Component({
    selector: 'language-select',
    template: `
        <div
            class="flex flex-col w-[18.5rem] rounded bg-white shadow relative -top-12 -right-1 dark:bg-neutral-700 dark:text-white"
            (click)="close()"
        >
            <div class="flex items-center space-x-2  p-2">
                <app-icon class="text-2xl">arrow_back</app-icon>
                <div class="" i18n>English</div>
            </div>
            <div class="text-xs opacity-60 mb-4 px-2 p-4" i18n>
                Select a language
            </div>
            <a
                [href]="'/workplace/' + lang.id + '/'"
                *ngFor="let lang of locales"
                class="w-full p-4 border-t border-gray-200"
            >
                {{ lang.name }} - {{ lang.flag }}
            </a>
        </div>
    `,
    styles: [``],
})
export class LanguageSelectComponent {
    public get locales() {
        return this._settings.get('app.locales') || [];
    }

    public readonly close = () => this._data?.close();

    constructor(
        @Inject(CustomTooltipData) private _data: any,
        private _settings: SettingsService
    ) {}
}
