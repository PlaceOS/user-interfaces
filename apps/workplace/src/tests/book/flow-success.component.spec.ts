import { createRoutingFactory, SpectatorRouting } from '@ngneat/spectator/jest';
import { IconComponent } from '@placeos/components';
import { MockComponent } from 'ng-mocks';

import { FlowSuccessComponent } from '../../app/book/flow-success.component';

describe('FlowSuccessComponent', () => {
    let spectator: SpectatorRouting<FlowSuccessComponent>;
    const createComponent = createRoutingFactory({
        component: FlowSuccessComponent,
        declarations: [MockComponent(IconComponent)],
    });

    beforeEach(() => (spectator = createComponent()));

    it('should create component', () => {
        expect(spectator.component).toBeTruthy();
    });

    it('should match snapshot', () => {
        expect(spectator.element).toMatchSnapshot();
    });

    it('should link to Schedule', () => {
        expect('a').toContainText('My Day');
    });

    it('should link to new booking', () => {
        expect('a').toContainText('New Booking');
    });
});
