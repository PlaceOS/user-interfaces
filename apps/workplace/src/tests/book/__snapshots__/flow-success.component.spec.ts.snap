// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlowSuccessComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="fixed inset-0 flex flex-col bg-secondary items-center justify-center z-50"
    name="success"
  >
    <div
      class="text-4xl rounded-full bg-white text-success mb-4"
    >
      <app-icon>
        done
      </app-icon>
    </div>
    <div
      class="text-center text-lg text-white mb-4"
    >
       Thank you, you space booking was successful!
      <br />
       An event has been added to your calendar 
    </div>
    <div
      class="flex items-center space-x-2"
    >
      <a
        button=""
        class="w-32"
        mat-button=""
      >
         New Booking 
      </a>
      <a
        button=""
        class="w-32"
        mat-button=""
      >
         My Day 
      </a>
    </div>
  </div>
</div>
`;
